### DISCUSSION TITLE
Select dropdown for keys option in `key-value` field

### DISCUSSION THREAD

**Discussion by rama-adi** [filamentphp/filament#259] [Ideas]
It would be great if we could select the keys from predefined options list. example:
```php
KeyValue::make('socials')
	->options(['facebook', 'twitter']);
```

**Comment by danharrin** [comment]
I think @ryangjchandler had this idea the other day too!

**Reply by fauzie811** [reply]
The idea didn't make it to v3 I see.

**Comment by madsem** [comment]
I was looking for the same functionality, it seems possible by using the `default` method:

```php
Forms\Components\KeyValue::make('api_authentication')
                            ->default(fn($state) => [
                                "username" => '',
                                "password" => '',
                            ])
                            ->keyLabel('Type')
                            ->valueLabel('Value')
                            ->disableAddingRows()
                            ->disableDeletingRows()
                            ->disableEditingKeys(),

                    ]),
```
<img width="611" alt="Screen Shot 2022-04-14 at 7 29 44 PM" src="https://user-images.githubusercontent.com/3166991/163442190-c08171fd-5095-46d4-97bd-e0a36306b481.png">


is that a good way to go about it, or are there any weird side effects to be expected @danharrin  ? :D

**Reply by danharrin** [reply]
Nope that's correct! Along with 

```php
->disableAddingRows()
->disableDeletingRows()
->disableEditingKeys()
```

**Reply by rajatpatelz** [reply]
@danharrin How Can I achieve this in Filament v3.
`->default()` method doesn't seems to work.

**Reply by underdpt** [reply]
Yes, it would be good to have this funcionality on Filament v3.

**Reply by GL1TCH1337** [reply]
Is it possible with v3?

**Reply by nicolawebdev** [reply]
+1 for v3

**Reply by leandrocfe** [reply]
`default()` in V3

```php
Forms\Components\KeyValue::make('api_authentication')
    ->default(fn($state) => [
        "username" => 'xx',
        "password" => 'yy',
    ])
    ->keyLabel('Type')
    ->valueLabel('Value')
    ->addable(false)
    ->deletable(false)
    ->editableKeys(false)
```

_This only works on Create Pages, as Edit Pages will always fill the data from the model_
